pkg/api/handlers/compat/images_build.go

```diff
@@ -224,8 +225,14 @@ func BuildImage(w http.ResponseWriter, r *http.Request) {
 				// it's not json, assume just a string
 				m = []string{query.Dockerfile}
 			}
+
 			for _, containerfile := range m {
-				containerFiles = append(containerFiles, filepath.Join(contextDirectory, filepath.Clean(filepath.FromSlash(containerfile))))
+				u, err := url.ParseRequestURI(containerfile)
```

Comment:
what happens if there IS an error?

Reply:
We ignore it and treat it as a file path, which is status quo.

tmp/Dockerfile was causing an error, while
/tmp/Dockerfile passes.  So we treat either case as a file.